`test_interpolate_by.test_interpolate_vs_numpy` fails often

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
pytest -vm "" tests/unit/operations/test_interpolate_by.py::test_interpolate_vs_numpy
```


### Log output

```shell
____________________________________ test_interpolate_vs_numpy ____________________________________
tests/unit/operations/test_interpolate_by.py:166: in test_interpolate_vs_numpy
    def test_interpolate_vs_numpy(data: st.DataObject, x_dtype: pl.DataType) -> None:
tests/unit/operations/test_interpolate_by.py:230: in test_interpolate_vs_numpy
    assert_series_equal(result, expected)
polars/_utils/deprecation.py:119: in wrapper
    return function(*args, **kwargs)
E   AssertionError: Series are different (value mismatch)
E   [left]:  [164583760.0, 1.7881393432617188e-07, 0.0]
E   [right]: [164583760.0, 1.65617217337068e-07, 0.0]
E   Falsifying example: test_interpolate_vs_numpy(
E       data=data(...),
E       x_dtype=Float64,  # or any other generated value
E   )
E   Draw 1: shape: (4, 2)
E   ┌────────────┬─────────────┐
E   │ ts         ┆ value       │
E   │ ---        ┆ ---         │
E   │ f64        ┆ f64         │
E   ╞════════════╪═════════════╡
E   │ -9.9968e14 ┆ 1.6458376e8 │
E   │ 0.0        ┆ null        │
E   │ 0.0        ┆ null        │
E   │ 1.0        ┆ 0.0         │
E   └────────────┴─────────────┘
E   Explanation:
E       These lines were always and only run by failing examples:
E           /xxx/py-polars/polars/testing/asserts/series.py:327
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.125.1', b'AXicc2R0ZGHQOMyTubKoZWuCBgMUIBj2HyAMR0YNx8XzNi0AMUGQEa6EAQCkrwpt') as a decorator on your test case
===================================== short test summary info =====================================
FAILED tests/unit/operations/test_interpolate_by.py::test_interpolate_vs_numpy - AssertionError: Series are different (value mismatch)
```

### Issue description

The test  `test_interpolate_vs_numpy`  at `tests/unit/operations/test_interpolate_by.py`fails sometimes. Likely due to fp precision issue. 
Maybe more lenient tolerance is needed for comparison against `np.interp` for floating point values.

### Expected behavior

Test result should be deterministic.

### Installed versions

<details>

```
--------Version info---------
Polars:              1.27.1
Index type:          UInt32
Platform:            macOS-14.4.1-arm64-arm-64bit-Mach-O
Python:              3.13.1 (main, Dec  3 2024, 17:59:52) [Clang 16.0.0 (clang-1600.0.26.4)]
LTS CPU:             False
```

</details>

@mcrumiller Yep I can reproduce: 
Output:
```
result: shape: (3,)
Series: 'value' [f64]
[
	2.9753327e8
	0.0
	0.0
]
expected: shape: (3,)
Series: 'value' [f64]
[
	2.9753327e8
	-1.4901e-8
	0.0
]
```

```
from polars.testing import assert_series_equal
assert_series_equal(result, expected)
```

Produces the exact error 

```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[24], line 2
      1 from polars.testing import assert_series_equal
----> 2 assert_series_equal(result, expected)

File /polars/py-polars/polars/_utils/deprecation.py:119, in deprecate_renamed_parameter.<locals>.decorate.<locals>.wrapper(*args, **kwargs)
    114 @wraps(function)
    115 def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
    116     _rename_keyword_argument(
    117         old_name, new_name, kwargs, function.__qualname__, version
    118     )
--> 119     return function(*args, **kwargs)

    [... skipping hidden 3 frame]

File /polars/py-polars/polars/testing/asserts/utils.py:12, in raise_assertion_error(objects, detail, left, right, cause)
     10 __tracebackhide__ = True
     11 msg = f"{objects} are different ({detail})\n[left]:  {left}\n[right]: {right}"
---> 12 raise AssertionError(msg) from cause

AssertionError: Series are different (value mismatch)
[left]:  [297533270.0, 0.0, 0.0]
[right]: [297533270.0, -1.4901161193847656e-08, 0.0]
```

The assertion fails at `assert_series_equal(result, expected, atol=1e-8)` but passes at `atol=1e7`

--- a/.github/workflows/test-coverage.yml
+++ b/.github/workflows/test-coverage.yml
@@ -93,6 +93,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: '3.12'
+          check-latest: true
 
       - name: Install uv
         run: |

--- a/py-polars/pyproject.toml
+++ b/py-polars/pyproject.toml
@@ -285,3 +285,13 @@ exclude_lines = [
   "if TYPE_CHECKING:",
   "from typing_extensions import ",
 ]
+
+[[tool.uv.index]]
+name = "pytorch"
+url = "https://download.pytorch.org/whl/cpu"
+explicit = true
+
+[tool.uv.sources]
+torch = [
+  { index = "pytorch", marker = "python_version < '3.1.3' or sys_platform != 'win32'" },
+]

--- a/py-polars/pyproject.toml
+++ b/py-polars/pyproject.toml
@@ -293,5 +293,5 @@ explicit = true
 
 [tool.uv.sources]
 torch = [
-  { index = "pytorch", marker = "python_version < '3.1.3' or sys_platform != 'win32'" },
+  { index = "pytorch" },
 ]

--- a/py-polars/requirements-ci.txt
+++ b/py-polars/requirements-ci.txt
@@ -2,7 +2,6 @@
 # Packages that we require for unit tests that run on CI
 # (installable via `make requirements-all`)
 # -------------------------------------------------------
---extra-index-url https://download.pytorch.org/whl/cpu
 torch
 jax[cpu]
 pyiceberg>=0.7.1

